### ISSUE TITLE
handle test clients behind a capsule

### ISSUE THREAD

**Issue by pcfe** [RedHatSatellite/soe-ci#76]
currently we do not wait for capsule sync to finish before trying to install and test.

so either we need to make it clear that the test machines must not be behind a capsule or wait with `buildtestvms.sh` until capsule syncs are finished.

In case of the latter, an crude but workable solution might be
`hammer --csv task list --search 'state = running and label =  Actions::Katello::ContentView::CapsuleGenerateAndSync' | tail -n +2`
run in a loop until we get 0 lines. But more elegant would be to actually wait on jyust the capsule sync we need.

@ericzolf comments?
I'll start poking at this once I know your preference. (just assign ticket to me after you commented)

**Comment by pcfe** [comment]
@ericzolf 
the non elegant solution would be along these lines
(plus the usual header that pulls in `common.sh` and checks for the needed variables

```
#!/bin/bash

# so we want to wait as logn as this returns one or more lines
# hammer --csv task list --search 'state = running and label =  Actions::Katello::ContentView::CapsuleGenerateAndSync' | tail -n +2

count_capsule_syncs_in_progress()
{
  _CAPSULE_SYNC_COUNT=$(ssh -q -l ${PUSH_USER} -i ${RSA_ID} ${SATELLITE} "hammer --csv task list --search 'state = running and label =  Actions::Katello::ContentView::CapsuleGenerateAndSync'" | tail -n +2 | wc -l)
}

info "Checking if there is a Capsule Sync in progress"
count_capsule_syncs_in_progress
if [[ ${_CAPSULE_SYNC_COUNT} -gt 0 ]]
then
  _CHEKING_SINCE=$(date)
  warn "${_CAPSULE_SYNC_COUNT} sync jobs to capsules in progress"
  while [[ ${_CAPSULE_SYNC_COUNT} -gt 0 ]]
  do
    info "Will check again in 1 minute (at $(date --date "1 minute")"
    info "We have been checking since      ${_CHEKING_SINCE})"
    sleep 60
    count_capsule_syncs_in_progress
    warn "${_CAPSULE_SYNC_COUNT} sync jobs to capsules still running"
  done
else
  info "No sync jobs to capsules in progress"
fi
```

**Comment by pcfe** [comment]
note to self: the above worked well last night at $current_customer

Still will await @ericzolf input before deciding how I fix this upstream

**Comment by nstrug** [comment]
Fixed in #78
